src/general.c

```diff
@@ -66,6 +67,67 @@ char *get_input(void) {
   return output;
 }
 
+BYTE cgetn_input(char *s, BYTE len) {
+  memset(s, '\0', len);
+  get_input_terminated_by(INPUT_TERMINATOR_RETURN, s, len);
+  return strlen(s);
+}
+
+int cget_number(int default_val) {
+  char c;
+  int x, y;
+  uint8_t num_chars = 0;
+  char buf[6] = {0, 0, 0, 0, 0, 0};
+  int i = 0;
+  x = wherex();
+  y = wherey();
+  while (1) {
+    /* end of buffer? */
+    if (i == (sizeof(buf) - 1))
+      break;
+
+    c = cgetc();
+    /* Enter */
+    if (c == PETSCII_CR)
+      break;
+    /* Backspace */
+    if (c == PETSCII_DEL) {
+      if (i == 0)
+        continue;
+      buf[--i] = '\0';
+      cputcxy(x + i, y, ' ');
+      gotoxy(x + i, y);
+      continue;
+    }
+    /* Abort */
+    if (c == PETSCII_ESC) {
+      return default_val;
+    }
+    if (c == '-' && i == 0) {
+      buf[0] = c;
+      ++num_chars;
+      cputc(c);
+    } else if (c >= PETSCII_0 && c <= PETSCII_9) {
+      buf[i] = c;
+      ++i;
+      ++num_chars;
+      cputc(c);
+    }
+  }
+  if (num_chars == 0) {
+    return default_val;
+  } else if ((num_chars == 1) && (c == '-')) {
+    return default_val;
+  }
+  return atoi(buf);
+}
+
+void cget_return() {
```

Comment:
AFAICT, this discards input until return is pressed? Can you add a comment please?